### PR TITLE
This should fix a lock-order-inversion.

### LINKED ISSUE #12869: test: thread sanitizer reports lock-order-inversions in test_fipsinstall
The gcc thread sanitizer reports several lock order inversions in test_fipsinstall on current master (commit ec0ce188f44b7ab261b1d691e34913b338479b1f).

To reproduce the issue, build 

```
./config --debug --prefix=/opt/openssl-dev --sanitize=thread
make -j 16
```

(see [configdata.pm](https://github.com/openssl/openssl/files/5215258/configdata.pm.txt)) and run the fipsinstall test as follows:

```
make -j 16 tests V=1 TESTS=test_fipsinstall TSAN_OPTIONS=second_deadlock_stack=1
```

The following shows an excerpt of the output. (For the full log, see [test_fipsinstall.log](https://github.com/openssl/openssl/files/5215256/test_fipsinstall.log).)

```
make depend && make _tests
make: *** No rule to make target 'test_fipsinstall'.  Stop.
make: *** Waiting for unfinished jobs....
make[1]: Entering directory '/home/msp/src/openssl-master'
make[1]: Leaving directory '/home/msp/src/openssl-master'
make[1]: Entering directory '/home/msp/src/openssl-master'
( SRCTOP=. \
  BLDTOP=. \
  PERL="/usr/bin/perl" \
  FIPSKEY="f4556650ac31d35461610bac4ed81b1a181b2d8a43ea2854cbae22ca74560813" \
  EXE_EXT= \
  /usr/bin/perl ./test/run_tests.pl test_fipsinstall )
03-test_fipsinstall.t .. 
# The results of this test will end up in test-runs/test_fipsinstall
1..23
fipsinstall: Use -help for summary.
../../util/wrap.pl ../../apps/openssl fipsinstall -out fips.cnf -module -provider_name fips -macopt 'digest:SHA256' -macopt 'hexkey:f4556650ac31d35461610bac4ed81b1a181b2d8a43ea2854cbae22ca74560813' -section_name fips_sect => 1
ok 1 - fipsinstall fail
==================
WARNING: ThreadSanitizer: lock-order-inversion (potential deadlock) (pid=32437)
  Cycle in lock order graph: M38 (0x7b10000005c0) => M62 (0x7b1000000800) => M70 (0x7b10000006c0) => M38

  Mutex M62 acquired here while holding mutex M38 in main thread:
    #0 pthread_rwlock_wrlock /var/tmp/portage/sys-devel/gcc-9.3.0-r1/work/gcc-9.3.0/libsanitizer/tsan/tsan_interceptors.cc:1324 (libtsan.so.0+0x2f2ba)
    #1 CRYPTO_THREAD_write_lock crypto/threads_pthread.c:86 (libcrypto.so.3+0x2d5637)
    #2 ossl_provider_find crypto/provider_core.c:229 (libcrypto.so.3+0x2d0e0f)
    #3 provider_conf_load crypto/provider_conf.c:118 (libcrypto.so.3+0x2cfeec)
    #4 provider_conf_init crypto/provider_conf.c:169 (libcrypto.so.3+0x2d0135)
    #5 module_init crypto/conf/conf_mod.c:373 (libcrypto.so.3+0x180179)
    #6 module_run crypto/conf/conf_mod.c:231 (libcrypto.so.3+0x17fa86)
    #7 CONF_modules_load crypto/conf/conf_mod.c:136 (libcrypto.so.3+0x17f6f3)
    #8 CONF_modules_load_file_with_libctx crypto/conf/conf_mod.c:177 (libcrypto.so.3+0x17f82c)
    #9 CONF_modules_load_file crypto/conf/conf_mod.c:194 (libcrypto.so.3+0x17f8d3)
    #10 openssl_config_int crypto/conf/conf_sap.c:63 (libcrypto.so.3+0x180e59)
    #11 ossl_init_config crypto/init.c:237 (libcrypto.so.3+0x2c0449)
    #12 ossl_init_config_ossl_ crypto/init.c:235 (libcrypto.so.3+0x2c03f8)
    #13 pthread_once /var/tmp/portage/sys-devel/gcc-9.3.0-r1/work/gcc-9.3.0/libsanitizer/tsan/tsan_interceptors.cc:1405 (libtsan.so.0+0x2fabc)
    #14 CRYPTO_THREAD_run_once crypto/threads_pthread.c:126 (libcrypto.so.3+0x2d5715)
    #15 OPENSSL_init_crypto crypto/init.c:544 (libcrypto.so.3+0x2c0d88)
    #16 OPENSSL_init_ssl ssl/ssl_init.c:126 (libssl.so.3+0x3fa36)
    #17 apps_startup apps/openssl.c:67 (openssl+0x765b7)
    #18 main apps/openssl.c:255 (openssl+0x766ff)

  Mutex M38 previously acquired by the same thread here:
    #0 pthread_rwlock_wrlock /var/tmp/portage/sys-devel/gcc-9.3.0-r1/work/gcc-9.3.0/libsanitizer/tsan/tsan_interceptors.cc:1324 (libtsan.so.0+0x2f2ba)
    #1 CRYPTO_THREAD_write_lock crypto/threads_pthread.c:86 (libcrypto.so.3+0x2d5637)
    #2 OPENSSL_init_crypto crypto/init.c:542 (libcrypto.so.3+0x2c0d5e)
    #3 OPENSSL_init_ssl ssl/ssl_init.c:126 (libssl.so.3+0x3fa36)
    #4 apps_startup apps/openssl.c:67 (openssl+0x765b7)
    #5 main apps/openssl.c:255 (openssl+0x766ff)

  Mutex M70 acquired here while holding mutex M62 in main thread:
    #0 pthread_rwlock_wrlock /var/tmp/portage/sys-devel/gcc-9.3.0-r1/work/gcc-9.3.0/libsanitizer/tsan/tsan_interceptors.cc:1324 (libtsan.so.0+0x2f2ba)
    #1 CRYPTO_THREAD_write_lock crypto/threads_pthread.c:86 (libcrypto.so.3+0x2d5637)
    #2 ossl_property_write_lock crypto/property/property.c:127 (libcrypto.so.3+0x31a9e0)
    #3 ossl_method_store_add crypto/property/property.c:249 (libcrypto.so.3+0x31b031)
    #4 put_evp_method_in_store crypto/evp/evp_fetch.c:168 (libcrypto.so.3+0x277d7e)
    #5 ossl_method_construct_this crypto/core_fetch.c:95 (libcrypto.so.3+0x2bb77b)
    #6 algorithm_do_this crypto/core_algorithm.c:65 (libcrypto.so.3+0x2bb0b8)
    #7 provider_forall_loaded crypto/provider_core.c:673 (libcrypto.so.3+0x2d2718)
    #8 ossl_provider_forall_loaded crypto/provider_core.c:751 (libcrypto.so.3+0x2d2908)
    #9 ossl_algorithm_do_all crypto/core_algorithm.c:109 (libcrypto.so.3+0x2bb2e2)
    #10 ossl_method_construct crypto/core_fetch.c:124 (libcrypto.so.3+0x2bb94b)
    #11 inner_evp_generic_fetch crypto/evp/evp_fetch.c:316 (libcrypto.so.3+0x278408)
    #12 evp_generic_fetch crypto/evp/evp_fetch.c:363 (libcrypto.so.3+0x27866e)
    #13 EVP_MAC_fetch crypto/evp/mac_meth.c:155 (libcrypto.so.3+0x28e434)
    #14 fipsinstall_main apps/fipsinstall.c:415 (openssl+0x6b27b)
    #15 do_cmd apps/openssl.c:406 (openssl+0x76eb8)
    #16 main apps/openssl.c:288 (openssl+0x76888)

  Mutex M62 previously acquired by the same thread here:
    #0 pthread_rwlock_rdlock /var/tmp/portage/sys-devel/gcc-9.3.0-r1/work/gcc-9.3.0/libsanitizer/tsan/tsan_interceptors.cc:1294 (libtsan.so.0+0x2ef7a)
    #1 CRYPTO_THREAD_read_lock crypto/threads_pthread.c:73 (libcrypto.so.3+0x2d55f4)
    #2 ossl_provider_forall_loaded crypto/provider_core.c:744 (libcrypto.so.3+0x2d28e3)
    #3 ossl_algorithm_do_all crypto/core_algorithm.c:109 (libcrypto.so.3+0x2bb2e2)
    #4 ossl_method_construct crypto/core_fetch.c:124 (libcrypto.so.3+0x2bb94b)
    #5 inner_evp_generic_fetch crypto/evp/evp_fetch.c:316 (libcrypto.so.3+0x278408)
    #6 evp_generic_fetch crypto/evp/evp_fetch.c:363 (libcrypto.so.3+0x27866e)
    #7 EVP_MAC_fetch crypto/evp/mac_meth.c:155 (libcrypto.so.3+0x28e434)
    #8 fipsinstall_main apps/fipsinstall.c:415 (openssl+0x6b27b)
    #9 do_cmd apps/openssl.c:406 (openssl+0x76eb8)
    #10 main apps/openssl.c:288 (openssl+0x76888)

  Mutex M38 acquired here while holding mutex M70 in main thread:
    #0 pthread_rwlock_wrlock /var/tmp/portage/sys-devel/gcc-9.3.0-r1/work/gcc-9.3.0/libsanitizer/tsan/tsan_interceptors.cc:1324 (libtsan.so.0+0x2f2ba)
    #1 CRYPTO_THREAD_write_lock crypto/threads_pthread.c:86 (libcrypto.so.3+0x2d5637)
    #2 OPENSSL_init_crypto crypto/init.c:542 (libcrypto.so.3+0x2c0d5e)
    #3 ossl_ctx_global_properties crypto/property/property.c:103 (libcrypto.so.3+0x31a865)
    #4 ossl_method_store_fetch crypto/property/property.c:360 (libcrypto.so.3+0x31b644)
    #5 get_evp_method_from_store crypto/evp/evp_fetch.c:130 (libcrypto.so.3+0x277bf2)
    #6 ossl_method_construct crypto/core_fetch.c:130 (libcrypto.so.3+0x2bb98b)
    #7 inner_evp_generic_fetch crypto/evp/evp_fetch.c:316 (libcrypto.so.3+0x278408)
    #8 evp_generic_fetch crypto/evp/evp_fetch.c:363 (libcrypto.so.3+0x27866e)
    #9 EVP_MAC_fetch crypto/evp/mac_meth.c:155 (libcrypto.so.3+0x28e434)
    #10 fipsinstall_main apps/fipsinstall.c:415 (openssl+0x6b27b)
    #11 do_cmd apps/openssl.c:406 (openssl+0x76eb8)
    #12 main apps/openssl.c:288 (openssl+0x76888)

  Mutex M70 previously acquired by the same thread here:
    #0 pthread_rwlock_rdlock /var/tmp/portage/sys-devel/gcc-9.3.0-r1/work/gcc-9.3.0/libsanitizer/tsan/tsan_interceptors.cc:1294 (libtsan.so.0+0x2ef7a)
    #1 CRYPTO_THREAD_read_lock crypto/threads_pthread.c:73 (libcrypto.so.3+0x2d55f4)
    #2 ossl_property_read_lock crypto/property/property.c:122 (libcrypto.so.3+0x31a989)
    #3 ossl_method_store_fetch crypto/property/property.c:351 (libcrypto.so.3+0x31b5bd)
    #4 get_evp_method_from_store crypto/evp/evp_fetch.c:130 (libcrypto.so.3+0x277bf2)
    #5 ossl_method_construct crypto/core_fetch.c:130 (libcrypto.so.3+0x2bb98b)
    #6 inner_evp_generic_fetch crypto/evp/evp_fetch.c:316 (libcrypto.so.3+0x278408)
    #7 evp_generic_fetch crypto/evp/evp_fetch.c:363 (libcrypto.so.3+0x27866e)
    #8 EVP_MAC_fetch crypto/evp/mac_meth.c:155 (libcrypto.so.3+0x28e434)
    #9 fipsinstall_main apps/fipsinstall.c:415 (openssl+0x6b27b)
    #10 do_cmd apps/openssl.c:406 (openssl+0x76eb8)
    #11 main apps/openssl.c:288 (openssl+0x76888)

SUMMARY: ThreadSanitizer: lock-order-inversion (potential deadlock) crypto/threads_pthread.c:86 in CRYPTO_THREAD_write_lock
==================

... <stripped>

```

### PR DESCRIPTION
**Pull request by bernd-edlinger**
Fixes #12869

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by bernd-edlinger** [comment]
This is unfortunately not fixed:
```
WARNING: ThreadSanitizer: lock-order-inversion (potential deadlock) (pid=21190)
  Cycle in lock order graph: M38 (0x7b10000005c0) => M61 (0x7b1000000800) => M66 (0x7b1000000780) => M38

  Mutex M61 acquired here while holding mutex M38 in main thread:
    #0 pthread_rwlock_wrlock ../../../../gcc-trunk/libsanitizer/tsan/tsan_interceptors_posix.cpp:1368 (libtsan.so.0+0x4198e)
    #1 CRYPTO_THREAD_write_lock crypto/threads_pthread.c:86 (libcrypto.so.3+0x275a8a)
    #2 ossl_provider_find crypto/provider_core.c:229 (libcrypto.so.3+0x27281c)
    #3 provider_conf_load crypto/provider_conf.c:118 (libcrypto.so.3+0x271b01)
    #4 provider_conf_init crypto/provider_conf.c:169 (libcrypto.so.3+0x271b01)
    #5 module_init crypto/conf/conf_mod.c:378 (libcrypto.so.3+0x15fb4a)
    #6 module_run crypto/conf/conf_mod.c:237 (libcrypto.so.3+0x15fb4a)
    #7 CONF_modules_load crypto/conf/conf_mod.c:136 (libcrypto.so.3+0x15fb4a)
    #8 CONF_modules_load crypto/conf/conf_mod.c:89 (libcrypto.so.3+0x15fb4a)
    #9 CONF_modules_load_file_ex crypto/conf/conf_mod.c:179 (libcrypto.so.3+0x160769)
    #10 CONF_modules_load_file crypto/conf/conf_mod.c:200 (libcrypto.so.3+0x1608a8)
    #11 openssl_config_int crypto/conf/conf_sap.c:63 (libcrypto.so.3+0x160ba1)
    #12 ossl_init_config crypto/init.c:237 (libcrypto.so.3+0x265312)
    #13 ossl_init_config_ossl_ crypto/init.c:235 (libcrypto.so.3+0x265312)
    #14 pthread_once ../../../../gcc-trunk/libsanitizer/tsan/tsan_interceptors_posix.cpp:1449 (libtsan.so.0+0x447a0)
    #15 CRYPTO_THREAD_run_once crypto/threads_pthread.c:126 (libcrypto.so.3+0x275b2e)
    #16 OPENSSL_init_crypto crypto/init.c:544 (libcrypto.so.3+0x265adb)
    #17 OPENSSL_init_ssl ssl/ssl_init.c:126 (libssl.so.3+0x38a93)
    #18 apps_startup apps/openssl.c:66 (openssl+0x426980)
    #19 main apps/openssl.c:260 (openssl+0x426980)

  Mutex M38 previously acquired by the same thread here:
    #0 pthread_rwlock_wrlock ../../../../gcc-trunk/libsanitizer/tsan/tsan_interceptors_posix.cpp:1368 (libtsan.so.0+0x4198e)
    #1 CRYPTO_THREAD_write_lock crypto/threads_pthread.c:86 (libcrypto.so.3+0x275a8a)
    #2 OPENSSL_init_crypto crypto/init.c:542 (libcrypto.so.3+0x265ab5)
    #3 OPENSSL_init_ssl ssl/ssl_init.c:126 (libssl.so.3+0x38a93)
    #4 apps_startup apps/openssl.c:66 (openssl+0x426980)
    #5 main apps/openssl.c:260 (openssl+0x426980)

  Mutex M66 acquired here while holding mutex M61 in main thread:
    #0 pthread_rwlock_wrlock ../../../../gcc-trunk/libsanitizer/tsan/tsan_interceptors_posix.cpp:1368 (libtsan.so.0+0x4198e)
    #1 CRYPTO_THREAD_write_lock crypto/threads_pthread.c:86 (libcrypto.so.3+0x275a8a)
    #2 ossl_namemap_add_names crypto/core_namemap.c:280 (libcrypto.so.3+0x261d4a)
    #3 construct_evp_method crypto/evp/evp_fetch.c:191 (libcrypto.so.3+0x22df67)
    #4 ossl_method_construct_this crypto/core_fetch.c:70 (libcrypto.so.3+0x2613cb)
    #5 algorithm_do_this crypto/core_algorithm.c:65 (libcrypto.so.3+0x261241)
    #6 provider_forall_loaded crypto/provider_core.c:674 (libcrypto.so.3+0x2733ff)
    #7 ossl_provider_forall_loaded crypto/provider_core.c:752 (libcrypto.so.3+0x2733ff)
    #8 ossl_algorithm_do_all crypto/core_algorithm.c:109 (libcrypto.so.3+0x261369)
    #9 ossl_method_construct crypto/core_fetch.c:124 (libcrypto.so.3+0x261836)
    #10 inner_evp_generic_fetch crypto/evp/evp_fetch.c:316 (libcrypto.so.3+0x22e65f)
    #11 evp_generic_fetch crypto/evp/evp_fetch.c:363 (libcrypto.so.3+0x22e65f)
    #12 EVP_KEYMGMT_fetch crypto/evp/keymgmt_meth.c:213 (libcrypto.so.3+0x23b1a3)
    #13 int_ctx_new crypto/evp/pmeth_lib.c:262 (libcrypto.so.3+0x24b3fe)
    #14 EVP_PKEY_CTX_new_id crypto/evp/pmeth_lib.c:445 (libcrypto.so.3+0x24b3fe)
    #15 init_gen_str apps/genpkey.c:319 (openssl+0x447e28)
    #16 genpkey_main apps/genpkey.c:111 (openssl+0x4483ff)
    #17 do_cmd apps/openssl.c:411 (openssl+0x450e34)
    #18 main apps/openssl.c:293 (openssl+0x426bfe)

  Mutex M61 previously acquired by the same thread here:
    #0 pthread_rwlock_rdlock ../../../../gcc-trunk/libsanitizer/tsan/tsan_interceptors_posix.cpp:1338 (libtsan.so.0+0x3ef9e)
    #1 CRYPTO_THREAD_read_lock crypto/threads_pthread.c:73 (libcrypto.so.3+0x275a5a)
    #2 ossl_provider_forall_loaded crypto/provider_core.c:745 (libcrypto.so.3+0x273397)
    #3 ossl_algorithm_do_all crypto/core_algorithm.c:109 (libcrypto.so.3+0x261369)
    #4 ossl_method_construct crypto/core_fetch.c:124 (libcrypto.so.3+0x261836)
    #5 inner_evp_generic_fetch crypto/evp/evp_fetch.c:316 (libcrypto.so.3+0x22e65f)
    #6 evp_generic_fetch crypto/evp/evp_fetch.c:363 (libcrypto.so.3+0x22e65f)
    #7 EVP_KEYMGMT_fetch crypto/evp/keymgmt_meth.c:213 (libcrypto.so.3+0x23b1a3)
    #8 int_ctx_new crypto/evp/pmeth_lib.c:262 (libcrypto.so.3+0x24b3fe)
    #9 EVP_PKEY_CTX_new_id crypto/evp/pmeth_lib.c:445 (libcrypto.so.3+0x24b3fe)
    #10 init_gen_str apps/genpkey.c:319 (openssl+0x447e28)
    #11 genpkey_main apps/genpkey.c:111 (openssl+0x4483ff)
    #12 do_cmd apps/openssl.c:411 (openssl+0x450e34)
    #13 main apps/openssl.c:293 (openssl+0x426bfe)

  Mutex M38 acquired here while holding mutex M66 in main thread:
    #0 pthread_rwlock_wrlock ../../../../gcc-trunk/libsanitizer/tsan/tsan_interceptors_posix.cpp:1368 (libtsan.so.0+0x4198e)
    #1 CRYPTO_THREAD_write_lock crypto/threads_pthread.c:86 (libcrypto.so.3+0x275a8a)
    #2 OPENSSL_init_crypto crypto/init.c:542 (libcrypto.so.3+0x265ab5)
    #3 OBJ_sn2nid crypto/objects/obj_dat.c:588 (libcrypto.so.3+0x28924b)
    #4 evp_pkey_name2type crypto/evp/p_lib.c:1015 (libcrypto.so.3+0x241d1c)
    #5 help_get_legacy_alg_type_from_keymgmt crypto/evp/pmeth_lib.c:162 (libcrypto.so.3+0x249f17)
    #6 do_name crypto/core_namemap.c:126 (libcrypto.so.3+0x261a38)
    #7 do_name crypto/core_namemap.c:123 (libcrypto.so.3+0x261a38)
    #8 doall_util_fn crypto/lhash/lhash.c:205 (libcrypto.so.3+0x25fb00)
    #9 OPENSSL_LH_doall_arg crypto/lhash/lhash.c:220 (libcrypto.so.3+0x25fb00)
    #10 lh_NAMENUM_ENTRY_doall_DOALL_NAMES_DATA crypto/core_namemap.c:129 (libcrypto.so.3+0x261c4f)
    #11 ossl_namemap_doall_names crypto/core_namemap.c:141 (libcrypto.so.3+0x261c4f)
    #12 evp_names_do_all crypto/evp/evp_fetch.c:554 (libcrypto.so.3+0x22f2cf)
    #13 EVP_KEYMGMT_names_do_all crypto/evp/keymgmt_meth.c:277 (libcrypto.so.3+0x23b378)
    #14 get_legacy_alg_type_from_keymgmt crypto/evp/pmeth_lib.c:169 (libcrypto.so.3+0x24b431)
    #15 int_ctx_new crypto/evp/pmeth_lib.c:277 (libcrypto.so.3+0x24b431)
    #16 EVP_PKEY_CTX_new_id crypto/evp/pmeth_lib.c:445 (libcrypto.so.3+0x24b431)
    #17 init_gen_str apps/genpkey.c:319 (openssl+0x447e28)
    #18 genpkey_main apps/genpkey.c:111 (openssl+0x4483ff)
    #19 do_cmd apps/openssl.c:411 (openssl+0x450e34)
    #20 main apps/openssl.c:293 (openssl+0x426bfe)

  Mutex M66 previously acquired by the same thread here:
    #0 pthread_rwlock_rdlock ../../../../gcc-trunk/libsanitizer/tsan/tsan_interceptors_posix.cpp:1338 (libtsan.so.0+0x3ef9e)
    #1 CRYPTO_THREAD_read_lock crypto/threads_pthread.c:73 (libcrypto.so.3+0x275a5a)
    #2 ossl_namemap_doall_names crypto/core_namemap.c:140 (libcrypto.so.3+0x261c33)
    #3 evp_names_do_all crypto/evp/evp_fetch.c:554 (libcrypto.so.3+0x22f2cf)
    #4 EVP_KEYMGMT_names_do_all crypto/evp/keymgmt_meth.c:277 (libcrypto.so.3+0x23b378)
    #5 get_legacy_alg_type_from_keymgmt crypto/evp/pmeth_lib.c:169 (libcrypto.so.3+0x24b431)
    #6 int_ctx_new crypto/evp/pmeth_lib.c:277 (libcrypto.so.3+0x24b431)
    #7 EVP_PKEY_CTX_new_id crypto/evp/pmeth_lib.c:445 (libcrypto.so.3+0x24b431)
    #8 init_gen_str apps/genpkey.c:319 (openssl+0x447e28)
    #9 genpkey_main apps/genpkey.c:111 (openssl+0x4483ff)
    #10 do_cmd apps/openssl.c:411 (openssl+0x450e34)
    #11 main apps/openssl.c:293 (openssl+0x426bfe)

SUMMARY: ThreadSanitizer: lock-order-inversion (potential deadlock) crypto/threads_pthread.c:86 in CRYPTO_THREAD_write_lock
```

File: crypto/property/property.c
Comment (mspncp) [inline crypto/property/property.c:341]: The `#ifndef FIPS_MODULE` seems to be essential here. Without it, I get a linker error.
```
/usr/lib/gcc/x86_64-pc-linux-gnu/9.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: providers/libfips.a(libfips-lib-property.o): in function `ossl_method_store_fetch':
/home/msp/src/openssl/crypto/property/property.c:340: undefined reference to `OPENSSL_init_crypto'
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:23204: providers/fips.so] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/home/msp/src/openssl'
make: *** [Makefile:3208: build_sw] Error 2
```

Comment (bernd-edlinger) [reply crypto/property/property.c]: right.

**Review by paulidale — APPROVED** [review]

**Review by mspncp — CHANGES_REQUESTED** [review]
It looks good at first sight, at least the lock order inversion seems to be fixed without breaking #12565. However, in view of all the previous discussion, in particular https://github.com/openssl/openssl/pull/13288#issuecomment-721910813 I would appreciate to see a little bit more explaining text in the commit message. (This is a change request.)

Also, to be sure I would like to wait for @mattcaswell's independent review before considering this pull request ready-to-merge.

**Review by mattcaswell — APPROVED** [review]
LGTM

**Review by mspncp — APPROVED** [review]

**Comment by bernd-edlinger** [comment]
Pushed to master 07af944

**Pull request closed without merging**
